libr/core/cmd_print.c

```diff
@@ -1318,6 +1315,24 @@ R_API void r_core_print_cmp(RCore *core, ut64 from, ut64 to) {
 	free (b);
 }
 
+static inline bool pdi_check_end(int nb_opcodes, int nb_bytes, int i, int j) {
+	if (nb_opcodes) {
+		if (nb_bytes) {
+			return j < nb_opcodes && i < nb_bytes;
+		} else {
+			return j < nb_opcodes;
+		}
+	} else {
+		return i < nb_bytes;
+	}
+}
+
+static inline const char *pdi_get_color(RCore *core, int offset, const char *fallback) {
+	const char *color = (char *) ((void *) &core->cons->pal + offset);
```

Comment:
maybe worth checking offset boundaries?

not sure if this thing will be portable, incrementing a pointer to select an index in an array looks really bad to me